### PR TITLE
Add interactive Help message for Slack platform

### PR DESCRIPTION
**Pull request by mszostok**
This depends on https://github.com/kubeshop/botkube/pull/693 which needs to be merged first.

## Description

Changes proposed in this pull request:

- Add interactive Help message for Slack platform

Implemented on this PR:
- [x] Help is shown when BotKube is installed and links to a Slack channel (first connection)
	- [x] public message shown for everyone only once, right after installation for configured channels (initial config)
	- [x] if the bot is invited to other channel, the message won't be printed
- [x] The help dialog shows the dialog as per the design
- [x] The buttons run the appropriate commands

## Testing

To run e2e test locally:

1. Create cluster: `k3d cluster create`
2. Checkout PR: `gh pr checkout 725`
3. Install BotKube: 
    ```bash
    export SLACK_BOT_TOKEN="xoxb-"
    export SLACK_APP_TOKEN="xapp-1-"
    
    helm upgrade botkube --install --namespace botkube ./helm/botkube --wait --create-namespace \
	    -f ./helm/botkube/e2e-test-values.yaml \
	    --set communications.default-group.slack.botToken="${SLACK_BOT_TOKEN}" \
	    --set communications.default-group.slack.appToken="${SLACK_APP_TOKEN}" \
	    --set image.repository=kubeshop/pr/botkube \
	    --set image.tag=725-PR \
	    --set image.pullPolicy="Always" \
	    --set analytics.disable=true
    ```
4. Run tests:
    ```bash
    export KUBECONFIG="config"
    export SLACK_BOT_NAME="botkube"
    export SLACK_TESTER_NAME="tester"
    export SLACK_ADDITIONAL_CONTEXT_MESSAGE="local"
    export SLACK_TESTER_APP_TOKEN="xoxb-"
    make test-integration
    ```

## TODO

In follow-up, I will:
- [ ] Help is shown when BotKube is installed and links to a Slack channel (first connection)
	- [ ] save state to not show it per app restart
- [ ] A user runs `@BotKube help`
- [ ] Print a command header (so we can easily correlate the output)
- [ ] Print a plaintext message for others
- [ ] Add source bindings customization

## Related issue(s)

- #683

### CODE REVIEW

File: pkg/bot/slack.go [hunk 1]
```diff
@@ -140,18 +136,49 @@ func (b *Slack) Start(ctx context.Context) error {
...
+				}
+
+				websocketClient.Ack(*event.Request)
+
+				switch callback.Type {
+				case slack.InteractionTypeBlockActions:
+					b.log.Debugf("Got block action %s", utils.StructDumper().Sdump(callback.ActionCallback.BlockActions))
+
+					if len(callback.ActionCallback.BlockActions) != 1 {
+						fmt.Println("ignoring as too many actions selected")
```
Comment (pkosiec) [inline pkg/bot/slack.go:163]: Use `b.log` and also change the message - `Ignoring as the number of actions is different from 1` or something like this. I know there shouldn't be the case that the len is 0, but...

File: pkg/bot/slack.go [hunk 2]
```diff
@@ -140,18 +136,49 @@ func (b *Slack) Start(ctx context.Context) error {
...
+						continue
+					}
+
+					act := callback.ActionCallback.BlockActions[0]
+					if strings.HasPrefix(act.ActionID, "url:") {
+						continue // skip the url actions
```
Comment (pkosiec) [inline pkg/bot/slack.go:169]: Should we debug log this as well?

File: pkg/bot/slack.go [hunk 3]
```diff
@@ -320,12 +348,37 @@ func (b *Slack) SendMessage(ctx context.Context, msg string) error {
...
+
+	return errs.ErrorOrNil()
+}
+
+// SendInteractiveMessage sends message with interactive sections to Slack channels.
+func (b *Slack) SendInteractiveMessage(ctx context.Context, msg interactive.Message) error {
+	errs := multierror.New()
+	for _, channel := range b.getChannels() {
+		channelName := channel.Name
+		b.log.Debugf(">> Sending message to channel %q: %+v", channelName, msg)
```
Comment (pkosiec) [inline pkg/bot/slack.go:362]: nit: I think we can get rid of `>>` in logs 🤔

**Review by pkosiec — COMMENTED** [review]
Code looks good! A few minor comments, now I'll test the solution locally and give a final approval.

File: pkg/bot/interactive/api.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package interactive
+
+import (
+	"context"
+
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+// Bot represents bidirectional integration.
+type Bot interface {
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/bot/interactive/api.go:10]: discussed f2f, we will move the helper method to `main.go` to solve that problem.

File: pkg/bot/slack.go (hunk 2 above)
Comment (mszostok) [reply pkg/bot/slack.go:169]: IMO not, it's a normal behavior and for now I don't see any value in debug message

File: pkg/bot/slack.go (hunk 3 above)
Comment (mszostok) [reply pkg/bot/slack.go:362]: It was there already, so for consistency with other logs I left them, but if we agree I will remove it in all places, OK?

**Pull request merged**